src/shipping/strategies/amazon-pay-v2/amazon-pay-v2-shipping-strategy.ts

```diff
@@ -19,8 +20,15 @@ export default class AmazonPayV2ShippingStrategy implements ShippingStrategy {
         private _shippingStrategyActionCreator: ShippingStrategyActionCreator
     ) {}
 
-    updateAddress(): Promise<InternalCheckoutSelectors> {
-        return Promise.resolve(this._store.getState());
+    updateAddress(address: AddressRequestBody, options?: ShippingRequestOptions): Promise<InternalCheckoutSelectors> {
+        const updateAddressRequestBody = {
+            ...this._store.getState().shippingAddress.getShippingAddress(),
+            customFields: address.customFields,
+        } as AddressRequestBody;
```

Comment:
instead of using `as`, maybe you could actually construct the payload properly?

```js
const { country, ... updateAddressRequestBody } = this._store.getState().shippingAddress.getShippingAddress();

this._consignmentActionCreator.updateAddress({
  ...updateAddressRequestBody,
  customFields: address.customFields,
}, options)
```